x-pack/test/rule_registry/spaces_only/tests/trial/lifecycle_executor.ts

```diff
@@ -0,0 +1,242 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { ElasticsearchClient, Logger, LogMeta } from 'kibana/server';
+import sinon from 'sinon';
+import expect from '@kbn/expect';
+import { mappingFromFieldMap } from '../../../../../plugins/rule_registry/common/mapping_from_field_map';
+import {
+  AlertConsumers,
+  ALERT_REASON,
+  ALERT_UUID,
+} from '../../../../../plugins/rule_registry/common/technical_rule_data_field_names';
+import {
+  createLifecycleExecutor,
+  WrappedLifecycleRuleState,
+} from '../../../../../plugins/rule_registry/server/utils/create_lifecycle_executor';
+import type { FtrProviderContext } from '../../../common/ftr_provider_context';
+import {
+  Dataset,
+  IRuleDataClient,
+  RuleDataService,
+} from '../../../../../plugins/rule_registry/server';
+import {
+  MockRuleParams,
+  MockRuleState,
+  MockAlertContext,
+  MockAlertState,
+  MockAllowedActionGroups,
+} from '../../../common/types';
+import { AlertExecutorOptions } from '../../../../../plugins/alerting/server';
```

Comment:
If we want to keep using the correct terminology, this should be `import { AlertExecutorOptions as RuleExecutorOptions } from '../../../../../plugins/alerting/server';` instead